src/utils/FilterPeople.tsx

```diff
@@ -0,0 +1,65 @@
+import { useSearchParams } from 'react-router-dom';
+import { Person } from '../types';
+
+export const FilterPeople = (sourcePeople: Person[] | null) => {
+  const [searchParams] = useSearchParams();
+
+  if (sourcePeople === null) {
+    return null;
+  }
+
+  const sexFilter = searchParams.get('sex');
+  const query = searchParams.get('query');
+  const centuriesFilter = searchParams.getAll('centuries')
+    .map(century => (+century * 100));
+  const sortBy = searchParams.get('sort') as keyof Person;
+
+  let filteredPeople = [...sourcePeople];
+
+  if (sexFilter) {
+    filteredPeople = filteredPeople.filter(
+      person => person.sex === sexFilter,
+    );
+  }
+
+  if (query) {
+    const queryLC = query.toLowerCase();
+
+    filteredPeople = filteredPeople.filter(
+      person => (person.name.toLowerCase().includes(queryLC)
+      || person.motherName?.toLowerCase().includes(queryLC)
+      || person.name.toLowerCase().includes(queryLC)),
+    );
+  }
+
+  if (centuriesFilter.length) {
+    filteredPeople = filteredPeople.filter(
+      person => centuriesFilter.some(year => (
+        year >= person.born && (year - 100) < person.born
+      )),
+    );
+  }
+
+  if (sortBy) {
```

Comment:
It's better to split your logic into two functions, the first for filtering and the second for sorting. As it looks a bit complicated